src/main/java/com/google/sps/servlets/GroupsServlet.java

```diff
@@ -57,18 +58,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         ImmutableList.Builder<ImmutableMap<String, String>> groupsListBuilder = new ImmutableList.Builder<>();
         for (Entity group : preparedGroupsQuery.asIterable()) {
             ImmutableMap<String, String> groupMap = ImmutableMap.of(
+                GROUP_ID_PROPERTY, group.getKey().getName(),
                 GROUP_NAME_PROPERTY, (String) group.getProperty(GROUP_NAME_PROPERTY),
                 GROUP_IMAGE_PROPERTY, (String) group.getProperty(GROUP_IMAGE_PROPERTY),
                 GROUP_DESCRIPTION_PROPERTY, (String) group.getProperty(GROUP_DESCRIPTION_PROPERTY));
             groupsListBuilder.add(groupMap);
         }
         
         ImmutableList<ImmutableMap<String, String>> groupsList = groupsListBuilder.build();
+        ImmutableList<Long> userGroups = ImmutableList.copyOf(
+            ServletUtils.getGroupIdList(request.getUserPrincipal().getName()));
         
         // Each group has its own map which points to its info and all maps are passed into the template as a list
         // This will make it easier when groups are queried from Datastore
-        ImmutableMap<String, ImmutableList<ImmutableMap<String, String>>> groupsData = 
-            ImmutableMap.of(GROUPS_KEY, groupsList);
+        ImmutableMap<String, ImmutableList> groupsData = 
+            ImmutableMap.of(GROUPS_KEY, groupsList, USER_GROUPS, userGroups);
 
```

Comment:
Is this the map of the groups from getGroupsList(userEmail) or is this the new function you created?

Reply:
It's from the new function I made. basically it's just a list of all the group Ids the user has joined